boa_engine/src/bytecompiler/jump_control.rs

```diff
@@ -66,7 +66,7 @@ pub(crate) enum JumpRecordAction {
 pub(crate) enum JumpRecordKind {
     Break,
     Continue,
-    Return,
+    Return { return_value_on_stack: bool },
```

Comment:
Wouldn't this increase the size of the enum? It might be better to have a new variant

Reply:
[I doesn't](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=7ab6df9c9c5f8852b90bd2ba26bf33c1), since a bool only needs a bit it can probably sore it in the discriminant, but even if would increase the size, I think it would be fine since we only have one per `JumpRecord`.